packages/landing/scripts/deploy.js

```diff
@@ -33,55 +34,65 @@ async function deploy() {
-        await recurseDirectory(DistDir, async (filePath, isDir) => {
-            if (isDir) return true;
-            const fileData = await fs.readFile(`${DistDir}/${filePath}`);
-            const hash = crypto.createHash('sha512').update(fileData).digest('base64');
-
-            const oldHash = await getHash(s3BucketName, filePath);
-            // Only upload files if they have changed
-            if (oldHash == hash) {
-                console.log('Skipped', filePath, `${(fileData.byteLength / 1024).toFixed(2)}Kb`, hash);
-                return true;
-            }
-            hasChanges = true;
-
-            // Don't set cache control for index.html as it may change all other files are hashed and immutable
-            const cacheControl = filePath == 'index.html' ? undefined : 'public, max-age=604800, immutable';
-
-            console.log('Uploading', filePath, `${(fileData.byteLength / 1024).toFixed(2)}Kb`, hash);
-            const res = await s3
-                .putObject({
-                    Bucket: s3BucketName,
-                    Key: filePath,
-                    Body: fileData,
-                    Metadata: { [HashKey]: hash },
-                    ContentType: mime.contentType(filePath),
-                    CacheControl: cacheControl,
-                })
-                .promise();
-
-            console.log('\tDone', res.ETag);
+    let changes = [];
+
+    async function upload(filePath, fileData, ContentType, CacheControl, hash) {
+        changes.push(filePath);
+
+        console.log('Uploading', filePath, `${(fileData.byteLength / 1024).toFixed(2)}Kb`, hash);
+        const res = await s3
+            .putObject({
+                Bucket: s3BucketName,
+                Key: filePath,
+                Body: fileData,
+                Metadata: { [HashKey]: hash },
+                ContentType,
+                CacheControl,
+            })
+            .promise();
+        console.log('\tDone', res.ETag);
+    }
+
+    // Upload all files from the dist directory, including any sub directories, to S3 bucket .
+    await recurseDirectory(DistDir, async (filePath, isDir) => {
+        if (isDir) return true;
+        const fileData = await fs.readFile(`${DistDir}/${filePath}`);
+        const hash = crypto.createHash('sha512').update(fileData).digest('base64');
+
+        const oldHash = await getHash(s3BucketName, filePath);
+        // Only upload files if they have changed
+        if (oldHash == hash) {
+            console.log('Skipped', filePath, `${(fileData.byteLength / 1024).toFixed(2)}Kb`, hash);
             return true;
-        });
-    };
+        }
+        const contentType = mime.contentType(extname(filePath));
 
-    await uploadDirectory();
+        // Don't set cache control for index.html as it may change all other files are hashed and immutable
+        const cacheControl = filePath.endsWith('.html') ? undefined : 'public, max-age=604800, immutable';
```

Comment:
Comment needs to change to "...for html files as they may change..."

I was wondering if we should set cache control for html files to `public, max-age=300, stale-while-revalidate=86400` which is fresh for 5 minutes then use max one day but reload then must reload.

Reply:
I don't think the `index.html` gets hit that often so I dont think its a huge issue to keep it somewhat fresh.

Quickly reading the docs on `stale-while-revalidate` I think It would be best to keep the max-age quite low eg `60` seconds or less and maybe keep the `stale-while-revalidate` at say 5 minutes?

AKA I dont want to wait up to 5 minutes for a index.html change to go through.

Reply:
I am changing the logic so that all files containing a version number are marked immutable `public, max-age=604800, immutable` and all other files have a short lived cache control `public, max-age=60, stale-while-revalidate=300`

Only `index.html` will be invalidated from the cache. Arguably this is not needed either.